src/xtopcom/xunit_service/src/xbatch_packer.cpp

```diff
@@ -423,6 +429,12 @@ bool xbatch_packer::reset_xip_addr(const xvip2_t & new_addr) {
     return xcsaccount_t::reset_xip_addr(new_addr);
 }
 
+bool xbatch_packer::set_fade_xip_addr(const xvip2_t & new_addr) {
+    xdbg("xbatch_packer::set_fade_xip_addr set fade xip from %s to %s", xcons_utl::xip_to_hex(m_faded_xip2).c_str(), xcons_utl::xip_to_hex(new_addr).c_str());
+    m_faded_xip2 = new_addr;
```

Comment:
Will it happen that 2 role faded at the same time?

Reply:
Now , validator && auditor wouldn't be elect in same shard, in other word, won't work in same table in one physical node.
We should seperate adv && con roles into different batch_packer.